File: common-rule-engine/src/test/files/checks/SwitchDefaultLocationCheck/SwitchDefaultLocationCheck.java
import java.time.DayOfWeek;

class Foo {
  void foo(MyEnum myEnum, DayOfWeek dow) {
    switch (0) { // Compliant
    }

    switch (0) { // Compliant
      case 0:
    }

    switch (0) { // Compliant
      default:
    }

    switch (0) {
      default:   // Compliant
      case 0:
    }

    switch (myEnum) { // Compliant
      case A:
        break;
      case B:
        break;
    }

    switch (myEnum) { // Compliant
      case A:
      case B:
      case C:
        break;
    }

    switch (myEnum) { // Compliant
      case A:
      case B:
        break;
      default:
        break;
    }

    switch (dow) { // Compliant
      case FRIDAY:
        break;
      case MONDAY:
        break;
    }

    switch (myEnum) { // Compliant
      case A:
      case B:
        switch (dow) { // Compliant
          case FRIDAY:
            break;
          default:
            break;
        }
        break;
    }

    switch (myEnum) {
      case A:
        break;
      default:  // Noncompliant {{Move this "default" case clause to the beginning or end of this "switch" statement.}}
        break;
      case B:
        break;
    }

    switch (myEnum) { // Compliant
      case A:
        switch (dow) { // Compliant
          case FRIDAY:
            break;
          default:
            break;
        }
        break;
      case B:
        break;
      default:
    }
  }
}

enum MyEnum {
  A, B, C;

  MyEnum field;
}

File: common-rule-engine/src/test/files/checks/SwitchDefaultLocationCheck/SwitchDefaultLocationCheck.go
package main

type f struct {
	a, b, c int
}

func foo(tag int, x interface{}) {
	switch tag {
	default: // Compliant
		bar()
	case 0, 1, 2, 3:
		bar()
	case 4, 5, 6, 7:
		bar()
	}


	switch tag {
	case 0, 1, 2, 3:
		bar()
	default:  // Noncompliant {{Move this "default" case clause to the beginning or end of this "switch" statement.}}
		bar()
	case 4, 5, 6, 7:
		bar()
	}

	switch tag { // Compliant
	case 0, 1, 2, 3:
		bar()
	case 4, 5, 6, 7:
		bar()
	}

	switch tag { // Compliant
	case 0, 1, 2, 3:
		bar()
	case 4, 5, 6, 7:
		bar()
	default:
		bar()
	}

	switch tag { // Compliant
	case 0, 1, 2, 3:
		bar()
	case 4, 5, 6, 7:
		switch tag {  // Compliant
		case 5:
			bar()
		case 6:
			bar()
		}
	default:
		bar()
	}

	switch x.(type) { // Compliant
	case nil:
		bar("x is nil")
	case bool, string:
		bar("type is bool or string")
	}

	switch x.(type) {
	case nil:
		bar("x is nil")
	default:  // Noncompliant {{Move this "default" case clause to the beginning or end of this "switch" statement.}}
		bar("don't know the type")
	case bool, string:
		bar("type is bool or string")
	}

	switch x.(type) {
	default:  // Compliant
		bar("don't know the type")
	case nil:
		bar("x is nil")
	case bool, string:
		bar("type is bool or string")
	}

	switch x.(type) { // Compliant
	case nil:
		bar("x is nil")
	case bool, string:
		bar("type is bool or string")
	default:
		bar("don't know the type")
	}
}

func bar(s ... string) {
}
